src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -4648,6 +4647,14 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
             image_version.safety_profile = GalleryImageVersionSafetyProfile(
                 allow_deletion_of_replicated_locations=allow_replicated_location_deletion)
     else:
+        source = GalleryArtifactSource(managed_image=ManagedArtifact(id=managed_image))
+        profile = ImageVersionPublishingProfile(exclude_from_latest=exclude_from_latest,
+                                                end_of_life_date=end_of_life_date,
+                                                target_regions=target_regions or [TargetRegion(name=location)],
+                                                source=source, replica_count=replica_count,
+                                                storage_account_type=storage_account_type)
+        if replication_mode is not None:
+            profile.replication_mode = replication_mode
```

Comment:
If we do https://github.com/Azure/azure-cli/pull/24709/files#r1031946802, the code here can be simplified as
```suggestion
        source = GalleryArtifactSource(managed_image=ManagedArtifact(id=managed_image))
        profile.source = source
```